exercises/src/test/scala/fpinscala/localeffects/LocalEffectsSpec.scala

```diff
@@ -3,5 +3,65 @@ package fpinscala.localeffects
 import org.specs2.mutable._
 
 class LocalEffectsSpec extends Specification {
+  "STArray" >> {
+    "fill" >> {
+
+      val runnable = new RunnableST[(Int, Int)] {
+        override def apply[S]: ST[S, (Int, Int)] = {
+          for {
+            a <- STArray(10, 1)
+            _ <- a.fill(Map(0 -> 5, 5 -> 6))
+            first <- a.read(0)
+            second <- a.read(1)
+          } yield {
+            (first, second)
+          }
+        }
+      }
+
+      ST.runST(runnable) === (5, 1)
+    }
+  }
+
+  "Immutable" >> {
+    "quicksort" >> {
+      def test(xs: List[Int]) = {
+        Immutable.quicksort(xs) === xs.sorted
```

Comment:
It still fails for a list of size 1. ;)

e.g. `isSorted(List(1))` will result in:

```
scala.MatchError: List(1) (of class scala.collection.immutable.$colon$colon)
  at $anonfun$1.apply(<console>:8)
  at $anonfun$1.apply(<console>:8)
  at scala.collection.Iterator$class.forall(Iterator.scala:762)
  at scala.collection.AbstractIterator.forall(Iterator.scala:1202)
  ... 43 elided
```